OGDL/Parser.swift

```diff
@@ -105,12 +105,12 @@ private func foldr<G: GeneratorType, Result>(inout generator: G, initial: Result
 	return generator.next().map { combine($0, foldr(&generator, initial, combine)) } ?? initial
 }
 
-private func | <T, U> (left: Parser<T>.Function, right: String -> U) -> Parser<Either<T, U>>.Function {
-	return left | { (right($0), $0) }
+private func | <C: CollectionType, T, U where C.Generator.Element: Equatable>(left: Parser<C, T>.Function, right: () -> U) -> Parser<C, Either<T, U>>.Function {
```

Comment:
I know it would be if the type of right was: `(C, C.Index) -> U` but then compiling takes endlessly.